apis/actions.md

```diff
@@ -0,0 +1,123 @@
+---
+description: Creating tool items, GLib.Actions, and keyboard shortcuts
+---
+
+# Actions
+
+GTK and GLib have a powerful API called [`GLib.Action`](https://valadoc.org/gio-2.0/GLib.Action.html) which can be used to define the primary actions of your app, assign them keyboard shortcuts, and tie them to [`Actionable`](https://valadoc.org/gtk+-3.0/Gtk.Actionable.html) widgets in your app like Buttons and Menu Items. In this section, we're going to create a Quit action for your app with an assigned keyboard shortcut and a Button that shows that shortcut in a tooltip. Begin by creating a `Gtk.Application` with a `Gtk.ApplicationWindow` as you've done in previous examples.
+
+## Gtk.HeaderBar
+
+Let's begin by creating a new [`Gtk.HeaderBar`](https://valadoc.org/gtk+-3.0/Gtk.HeaderBar.html). Typically your app will have a HeaderBar at the top of the window that will contain tool items, which users will interact with to trigger your app's actions.
+
+```vala
+protected override void activate () {
+    var headerbar = new Gtk.HeaderBar () {
+        show_close_button = true
+    };
+
+    var main_window = new Gtk.ApplicationWindow (this) {
+        default_height = 300,
+        default_width = 300,
+        title = "Actions"
+    };
+    main_window.set_titlebar (headerbar);
+    main_window.show_all ();
+}
+```
+
+Since we're using this HeaderBar as our app's main titlebar, we need to set `show_close_button` to `true` so that GTK knows to include window controls. We can then override our Window's built-in titlebar with the `set_titlebar ()` method.
+
+Now let's create a new [`Gtk.Button`](https://valadoc.org/gtk+-3.0/Gtk.Button.html) with a big colorful icon and add it to our headerbar:
```

Comment:
I am not sure it is clear that the following code *replaces* the first lines of the `activate ()` functions ( the first version produces a standard close button).

Reply:
Should be more clear now, let me know what you think!

Reply:
Yes, that is clearer.